### ISSUE TITLE
Inconsistent whitespace removal for operators in 1.8.15

### ISSUE THREAD

**Issue by swt2c** [doxygen/doxygen#6814]
In 1.8.15 (specifically since 449a7e2b4ff114a72be573013558bae19672ebbc), the whitespace removal for operators is inconsistent.  For example, it is removed for `operator +-` but not for `operator -=`.  To illustrate further, this code:

```
class MyClass {
 public:
  MyClass();
  MyClass& operator -=(int i);
  MyClass& operator +-(int i);
}
```
Generates this with 1.8.14:
```
$ grep operator xml.1.8.14/classMyClass.xml 
        <definition>MyClass&amp; MyClass::operator-=</definition>
        <name>operator-=</name>
        <definition>MyClass&amp; MyClass::operator+-</definition>
        <name>operator+-</name>
      <member refid="classMyClass_1abfd49f46c1949a730a81e20274b7594e" prot="public" virt="non-virtual"><scope>MyClass</scope><name>operator+-</name></member>
      <member refid="classMyClass_1a16e4a6149841a56169d787441b982eef" prot="public" virt="non-virtual"><scope>MyClass</scope><name>operator-=</name></member>
```
But with 1.8.15/master:
```
$ grep operator xml/classMyClass.xml 
        <definition>MyClass&amp; MyClass::operator -=</definition>
        <name>operator -=</name>
        <definition>MyClass&amp; MyClass::operator+-</definition>
        <name>operator+-</name>
      <member refid="classMyClass_1a344e8026c507036464111e99a7b7823c" prot="public" virt="non-virtual"><scope>MyClass</scope><name>operator -=</name></member>
      <member refid="classMyClass_1abfd49f46c1949a730a81e20274b7594e" prot="public" virt="non-virtual"><scope>MyClass</scope><name>operator+-</name></member>
```
The problematic operators appear to be (at least): `!=`, `-`, `-=`, `*`, `*=`, `[]`

**Comment by albert-github** [comment]
My personal opinion is that now the operators with the space (i.e. `-=` so the 'problematic operators') look better in e.g. HTML etc. but this is of course a matter of taste.

When looking at the "C++ Standard Draft Sources"  (https://github.com/cplusplus/draft in papers/n4800.pdf or source /overloading.tex) the overloading operator is written directly behind the word operator without any space.

The problem results from `util.cpp`, the function `QCString removeRedundantWhiteSpace(const QCString &s)` where in the `case` construct for ` `  the `g_charAroundSpace.charMap[(uchar)nc].after` for a `+` sign is set to `FALSE` (happens for some other characters as well like `/`) and for the other mentioned characters it is set by default to `TRUE` (and a space is inserted).

The `removeRedundantWhiteSpace` is used on a lot of places so I'm not sure about the fix (testing for the word operator?).

(Note the issue speaks of `+=` but the example has `+-` but the `+-` shows the same result as the `+=`.)

**Comment by swt2c** [comment]
I agree there could be differences opinion on the spacing, but unfortunately at least one project (that I'm aware of) relies on parsing the XML output, so it seems like maintaining the long-standing behavior would be the right thing to do.

In fact, this is worse than I originally thought.  In the cases where there is no space in the existing code, e.g. `operator*=`, doxygen is now adding spaces to the output.

I'll try to look at the code to see if I can identify a fix, hopefully one can be found that still resolves @mosra issue.

**Comment by doxygen** [comment]
I've commited a possible fix https://github.com/doxygen/doxygen/commit/e54ccc0e6eb12a0720860d863a89cb3f1a83f2a2. Let me know if it works.

**Comment by albert-github** [comment]
I did a quick check and it doesn't look bad, only the following gave some problems:
`*`, `*=`, `&` and `&&`
in 1.8.14 the `*` and `*=` were OK, the `&` and `&&` had a space.
Note that `&=` had a space in 1.8.14 and with this path it doesn't.

I used: [operator.zip](https://github.com/doxygen/doxygen/files/2825619/operator.zip)

**Comment by doxygen** [comment]
I've made a new commit for the * and & operators: https://github.com/doxygen/doxygen/commit/2802e2b4ee8158dba3f3584037e99907c6db7ec4

**Comment by albert-github** [comment]
As far as I can see the mentioned problems are solved.

**Comment by swt2c** [comment]
Looks good to me, thanks a lot Dimitri and albert-github!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
